nlp/src/main/java/hivemall/nlp/tokenizer/KuromojiUDF.java

```diff
@@ -170,6 +192,86 @@ private static CharArraySet stopWords(@Nonnull final String[] array)
         return results;
     }
 
+    @Nullable
+    private static UserDictionary userDictionary(@Nonnull final ObjectInspector oi)
+            throws UDFArgumentException {
+        if (HiveUtils.isConstListOI(oi)) {
+            return userDictionary(HiveUtils.getConstStringArray(oi));
+        } else if (HiveUtils.isConstString(oi)) {
+            return userDictionary(HiveUtils.getConstString(oi));
+        } else {
+            throw new UDFArgumentException(
+                "User dictionary MUST be given as an array of constant string or constant string (URL)");
+        }
+    }
+
+    @Nullable
+    private static UserDictionary userDictionary(@Nullable final String[] userDictArray)
+            throws UDFArgumentException {
+        if (userDictArray == null) {
+            return null;
+        }
+
+        StringBuilder builder = new StringBuilder();
+        for (String row : userDictArray) {
+            builder.append(row);
+            builder.append("\n");
+        }
+        InputStream is = new FastByteArrayInputStream(builder.toString().getBytes());
+        final Reader reader = new InputStreamReader(is);
+        try {
+            return UserDictionary.open(reader); // return null if empty
+        } catch (Throwable e) {
+            throw new UDFArgumentException(
+                "Failed to create user dictionary based on the given array<string>: " + e);
+        }
+    }
+
+    @Nullable
+    private static UserDictionary userDictionary(@Nullable final String userDictURL)
+            throws UDFArgumentException {
+        if (userDictURL == null) {
+            return null;
+        }
+
+        final HttpURLConnection conn;
+        try {
+            conn = HttpUtils.getHttpURLConnection(userDictURL);
+        } catch (Throwable e) {
```

Comment:
`Exception` is enough to catch both `IOException` and `IllegalArgumentException`